### PR TITLE
Cleanup or execute `Down` exactly once.

### PR DESCRIPTION
**Pull request by madhusudancs**
Fixes: Issue #1840

cc @kubernetes/test-infra-maintainers @shashidharatd

### CODE REVIEW

File: kubetest/e2e.go [hunk 1]
```diff
@@ -241,7 +247,15 @@ func run(deploy deployer) error {
...
 
 	if *down {
-		errs = appendError(errs, xmlWrap("TearDown", deploy.Down))
+		errs = appendError(errs, xmlWrap("TearDown", func() error {
+			if !downDone {
+				err := deploy.Down()
+				if err != nil {
+					return err
+				}
+				downDone := true
```
Comment (ixdy) [inline kubetest/e2e.go]: `=`, not `:=`

File: kubetest/e2e.go [hunk 2]
```diff
@@ -173,12 +176,15 @@ func run(deploy deployer) error {
...
 		// attempt to bring it down so we're not leaving resources around.
-		//
-		// TODO: We should try calling deploy.Down exactly once. Though to
-		// stop the leaking resources for now, we want to be on the safe side
-		// and call it explictly in defer if the other one is not called.
 		if *down {
-			defer xmlWrap("Deferred TearDown", deploy.Down)
+			defer xmlWrap("Deferred TearDown", func() error {
+				if !downDone {
+					err := deploy.Down()
```
Comment (ixdy) [inline kubetest/e2e.go]: since you're not setting `downDone` here, this could just be `return deploy.Down()`.

File: kubetest/e2e.go (hunk 1 above)
Comment (madhusudancs) [inline kubetest/e2e.go]: Oops, thanks for catching that. We need to go deeper: test-infra need tests :)

File: kubetest/e2e.go (hunk 2 above)
Comment (madhusudancs) [inline kubetest/e2e.go]: Done.

**Pull request merged**